src/integer_mod_q/poly_over_zq/from.rs

```diff
@@ -172,6 +172,105 @@ impl FromStr for PolyOverZq {
     }
 }
 
+impl<IntoZq: Into<Zq>> From<IntoZq> for PolyOverZq {
```

Comment:
IntoZq is kind of irritating to me and does not really fit to our naming (Integer, Rational, etc)

My suggestions:
- ResidueClass
- CongruenceClass
- IntegerModQ

Reply:
Selected `IntegerModQ`